pkg/apis/pulumi/v1alpha1/stack_types.go

```diff
@@ -75,7 +74,16 @@ type StackSpec struct {
 	//   * SSH private key (and it's optional password)
 	//   * Basic auth username and password
 	// Only 1 authentication path is valid.
+	// Deprecated. Use GitAuth instead.
 	GitAuthSecret string `json:"gitAuthSecret,omitempty"`
+
+	// (optional) GitAuth allows configuring git authentication options
+	// There are 3 different authentication options:
+	//   * Personal access token
+	//   * SSH private key (and it's optional password)
+	//   * Basic auth username and password
+	// Only 1 authentication mode is valid.
```

Comment:
```suggestion
	// Only one authentication mode can be set.
```

Side note: Why is this the case? Seems like we could support fallback options if multiple choices are set? (Actually, it looks like the code doesn't enforce this, so it would implicitly allow multiple options with a priority ordering of PAT/private key/basic auth)

Reply:
Unfortunately automation api doesn't really support that — only one of the auth models is allowed. I did miss enforcing the constraint though will do so now.

Reply:
I clarified the ordering and made it consistent.